### PR TITLE
sfp_api_design

### PR DESCRIPTION
**Pull request by keboliu**
create a SFP API design doc

### CODE REVIEW

File: doc/pmon/sfp_new_api.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# New SFP Platform API Design
+
+## 1. APIs for reading SFP info and DOM info
+
+Considering to migrate from reading eeprom directly via access sysfs to using ethtool, cause hw-mgmt provided sysfs symbols for SFP eeprom and not standard Linux practice, maybe obsoleted in the future.
```
Comment (andriymoroz-mlnx) [inline doc/pmon/sfp_new_api.md]: Are we sure all vendors have this implemented in their drivers? Or this is only upgrade for us?

Comment (keboliu) [reply doc/pmon/sfp_new_api.md]: This is only our own implementation, other vendors can still follow the old way, our specific implementation will not interfere with other vendor. But we need to persuade the community to agree to introduce 'ethtool' to  pmon, currently it only on the host.

File: doc/pmon/sfp_new_api.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+since this function will be called periodically, use ethtool to read out the raw data of the monitored items, then parse it with original sfputilbase.
+
+for example:
+
+    root@arc-mtbc-1001:/home/admin# ethtool -m sfp37 offset 0 length 10
+    Offset          Values
+    ------          ------
+    0x0000:         03 04 0b 00 00 00 00 40 08 00 
+
+### c. Check the present status of SFP
```
Comment (keboliu) [inline doc/pmon/sfp_new_api.md]: may can not ditingusih the case that bad eeprom and not present.

File: doc/pmon/sfp_new_api.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
+# New SFP Platform API Design
+
+## 1. APIs for reading SFP info and DOM info
```
Comment (keboliu) [inline doc/pmon/sfp_new_api.md]: performance

File: doc/pmon/sfp_new_api.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        # qsfp is not present
+    else
+        # qsfp is present
+    fi
+
+
+## 2. Set/Get lpmode, TX enable/disable functions
+
+### Solution 1.
+This is done by calling SX_APIs inside syncd, plan to extend the current mlnx_sfpd daemon inside syncd.
```
Comment (keboliu) [inline doc/pmon/sfp_new_api.md]: the implementation seems heavy,  what's the benefit can we get?

**Pull request closed without merging**